slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala

```diff
@@ -164,12 +173,197 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
       override def valueToSQLLiteral(v: Char) = stringJdbcType.valueToSQLLiteral(String.valueOf(v))
     }
 
+    /**
+      * Use [[ZonedDateTimeJdbcType]] or [[OffsetDateTimeJdbcType]] or [[LocalTimeJdbcType]]
+      * or [[OffsetTime]] or [[LocalDateTime]] or [[LocalDate]] or [[Instant]] instead.
+      */
+    @deprecated(message = "java.util.Date is deprecated, use java.time.LocalDateTime instead.", since = "3.2.0")
     class DateJdbcType extends DriverJdbcType[Date] {
       def sqlType = java.sql.Types.DATE
       def setValue(v: Date, p: PreparedStatement, idx: Int) = p.setDate(idx, v)
       def getValue(r: ResultSet, idx: Int) = r.getDate(idx)
       def updateValue(v: Date, r: ResultSet, idx: Int) = r.updateDate(idx, v)
-      override def valueToSQLLiteral(value: Date) = "{d '"+value.toString+"'}"
+      override def valueToSQLLiteral(value: Date) = s"{d '${value.toString}'}"
+    }
+
+    class LocalDateJdbcType extends DriverJdbcType[LocalDate] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = java.sql.Types.DATE
+      override def setValue(v: LocalDate, p: PreparedStatement, idx: Int) : Unit = {
+        p.setDate(idx, Date.valueOf(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalDate = {
+        r.getDate(idx) match {
+          case null => null
+          case date => date.toLocalDate
+        }
+      }
+      override def updateValue(v: LocalDate, r: ResultSet, idx: Int) : Unit = {
+        r.updateDate(idx, Date.valueOf(v))
+      }
+      override def valueToSQLLiteral(value: LocalDate) = {
+        s"'${value.toString}'"
+      }
+    }
+
+    class LocalTimeJdbcType extends DriverJdbcType[LocalTime] {
+      override def sqlType : Int = {
+        /**
+         * [[LocalTime]] will be persisted as a [[java.sql.Types.VARCHAR]] in order to
+         * avoid losing precision, because the ANSI SQL type ([[java.sql.Types.TIME]]) stores
+         * TIME with second precision, while [[LocalTime]] stores it with a millisecond one.
+         */
+        java.sql.Types.VARCHAR
```

Comment:
In my opinion, if you want to adjust perfectly to the database type you can still use the JDBC type, like `Timestamp` or `java.sql.Date`. If you are storing a `java.time.ZonedDateTime` you want to store the exact timezone, which is a functionality that is not available in any database supported by Slick, then you would expect some slow-down when querying these columns. 

On the other hand, `LocalTime` is implemented with the native type where available, so it will only be a `VARCHAR` in databases with the standard database implementation (With no milliseconds). 
This implementation is DB-agnostic, so you can change the database in the backend and all the behavior will be exactly the same.

Maybe I should add a table in the Slick documentation indicating which types are implemented natively in each database.